### PR TITLE
Logging improvements

### PR DESCRIPTION
**Pull request by cmdrkotori**
Turn the mechanism where logs can be pushed on to stderr into a full-blown feature.  This new feature is invoked with the argument `--log-to-console`.  This is principally a debugging feature.

**Why shouldn't I do this?** The log thread will flush stderr after every line, which may slow it down.

### CODE REVIEW

**Comment by tbertels** [comment]
Very nice!

I suggest rewording "Block the main thread while log flush completed" to "Block the main thread while log flush completes".

**Comment by cmdrkotori** [comment]
Done.
